### PR TITLE
[meson] add layer to +tools/

### PR DESCRIPTION
**Pull request by hexagonal-sun**
Add a new meson layer to the +tools layer set.  This basic layer is a simple
wrapper around meson mode.

### CODE REVIEW

File: layers/+tools/meson/packages.el [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+;; Copyright (c) 2012-2018 Sylvain Benner & Contributors
+;;
+;; Author: Matthew Leach <dev@mattleach.net>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;;; Commentary: A simple layer that wraps meson-mode
```
Comment (smile13241324) [inline layers/+tools/meson/packages.el:12]: Please remove the commentary header, this should go into the README.org if it is not already there.

File: layers/+tools/meson/packages.el [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+;;; Code:
```
Comment (smile13241324) [inline layers/+tools/meson/packages.el:14]: No `Code:` block in package files please.

File: layers/+tools/meson/packages.el [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+(defconst meson-packages
+  '(meson-mode))
+
+(defun meson/init-meson-mode ()
+  (use-package meson-mode
+    :defer t
+    :mode (("meson\\.build\\'" . meson-mode))))
+
+;;; packages.el ends here
```
Comment (smile13241324) [inline layers/+tools/meson/packages.el:24]: No `packages.el ends here` block in package files please.

File: layers/+tools/meson/packages.el [hunk 4]
```diff
@@ -0,0 +1,24 @@
+;;; packages.el --- meson layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2012-2018 Sylvain Benner & Contributors
```
Comment (smile13241324) [inline layers/+tools/meson/packages.el:3]: Please change the license header to mention 2020.
```suggestion
;; Copyright (c) 2012-2020 Sylvain Benner & Contributors
```

File: layers/+tools/meson/README.org [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+* Install
+To use this configuration layer, add it to your =~/.spacemacs=. You will need to
+add =meson= to the existing =dotspacemacs-configuration-layers= list in this
+file.
+
+* Key bindings
+
+| Key Binding | Description                              |
+|-------------+------------------------------------------|
+| ~F1~        | Lookup documentation for thing at point. |
```
Comment (tko) [inline layers/+tools/meson/README.org:28]: I know the <kbd>F1</kbd> comes from meson-mode itself but IMO spacemacs could use more spacemacs-y keybinding, along the lines of <kbd>SPC h d d</kbd> (seems free?) or <kbd>SPC a z d</kbd> (dash-at-point)

**Comment by hexagonal-sun** [comment]
@smile13241324 thanks for the review.  Your comments have been addressed and resolved.

File: layers/+tools/meson/README.org (hunk 5 above)
Comment (hexagonal-sun) [reply layers/+tools/meson/README.org:28]: Should this maybe be a leader key for the mode such as `SPC m h`?

Comment (hexagonal-sun) [reply layers/+tools/meson/README.org:28]: I only suggest as `f1` binds to `meson-lookup-doc-at-point` which seems major-mode specific.

Comment (smile13241324) [reply layers/+tools/meson/README.org:28]: I would also vote for a major mode specific key, per convention this would then be `SPC m h h`

**Comment by smile13241324** [comment]
Thank you for contributing to spacemacs 💜, I have cherry picked your commit into develop.

**Pull request closed without merging**